zebra-grpc/src/server.rs

```diff
@@ -67,6 +67,60 @@ where
 
         Ok(Response::new(reply))
     }
+
+    async fn clear_results(
+        &self,
+        request: tonic::Request<ClearResultsRequest>,
+    ) -> Result<Response<Empty>, Status> {
+        let keys = request.into_inner().keys;
+
+        if keys.is_empty() {
+            let msg = "must provide the keys for which to clear results";
+            return Err(Status::invalid_argument(msg));
+        }
+
+        let ScanServiceResponse::ClearedResults = self
+            .scan_service
+            .clone()
+            .ready()
+            .and_then(|service| service.call(ScanServiceRequest::ClearResults(keys)))
+            .await
+            .map_err(|err| Status::unknown(format!("scan service returned error: {err}")))?
+        else {
+            return Err(Status::unknown(
+                "scan service returned an unexpected response",
+            ));
+        };
+
+        Ok(Response::new(Empty {}))
```

Comment:
Do we want to return the removed results ?

Reply:
I think we should later accept an optional height range or a stop height for clearing only some of the results, but only return an empty response if it's successful or an error from this RPC, assuming that if they're clearing the results, they already have them or are no longer interested in them.

Reply:
ok, that makes sense, thank you.